common/src/services/auth.service.ts

```diff
@@ -115,11 +115,13 @@ export class AuthService implements AuthServiceAbstraction {
   }
 
   async logInTwoFactor(twoFactor: TokenRequestTwoFactor): Promise<AuthResult> {
-    try {
-      return await this.logInStrategy.logInTwoFactor(twoFactor);
-    } finally {
+    const result = await this.logInStrategy.logInTwoFactor(twoFactor);
+
+    // Only clear state if 2FA token has been accepted, otherwise we need to be able to try again
+    if (!result.requiresTwoFactor) {
       this.clearState();
     }
+    return result;
```

Comment:
We need to clear this state more aggressively than this.

Certainly we want a try/catch around this for unexpected errors, but I think we also want to somehow detect moving off of the 2fa page at all.

In some sense, I think a single-shot is the lesser of two evils.

Reply:
As discussed, have added:
* error handling in `logInTwoFactor` (not required in `logIn` because state has not yet been saved).
* a timeout that will clear state after 2 minutes, plus an error message that gets returned to the client